src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Compilation.cs

```diff
@@ -423,7 +423,43 @@ public GenericDictionaryLookup ComputeGenericLookup(MethodDesc contextMethod, Re
 
         // CoreCLR compat - referring to function pointer types handled as IntPtr. No MethodTable for function pointers for now.
         private static TypeDesc WithoutFunctionPointerType(TypeDesc type)
-            => type.IsFunctionPointer ? type.Context.GetWellKnownType(WellKnownType.IntPtr) : type;
+        {
+            switch (type.Category)
+            {
+                case TypeFlags.Array:
```

Comment:
Skip the parametrized type lookup if there is no change in the parameter - similar to how it is done for the generic case below?